### PR TITLE
add missing idxd-config module for qpl-deflate codec

### PR DESCRIPTION
**Pull request by jinjunzh**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Pull build description of idxd-config into a separate CMake file to avoid accidental removal in future.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 9fad9f234fce73bd5854e7d8a7ffffe6d2cca88a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49651/9fad9f234fce73bd5854e7d8a7ffffe6d2cca88a/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: contrib/idxd-config-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## accel_config is the utility library required by QPL-Deflate codec for controlling and configuring Intel® In-Memory Analytics Accelerator (Intel® IAA).
+option(ENABLE_ACCEL_CONFIG "Enable accel-config library" ${ENABLE_LIBRARIES})
+
+if (NOT ENABLE_QPL)
```
Comment (rschu1ze) [inline contrib/idxd-config-cmake/CMakeLists.txt:4]: Re my other comment about switching

```
add_contrib (qpl-cmake qpl)
add_contrib (idxd-config-cmake idxd-config)
```

... when doing so, access to ENABLE_QPL (l. 4) will no longer work then from here. Also, the accel-config build should rather not depend on `ENABLE_LIBRARIES` (l. 2), we should avoid a situation where accel-config is build but QPL isn't.

So maybe do this:
```cmake
# in contrib/CMakeLists:

[...]
if (OS_LINUX AND ARCH_AMD64 AND (ENABLE_AVX2 OR ENABLE_AVX512))
    option (ENABLE_QPL "Enable Intel® Query Processing Library" ${ENABLE_LIBRARIES})
elseif(ENABLE_QPL)
    message (${RECONFIGURE_MESSAGE_LEVEL} "QPL library is only supported on x86_64 arch with avx2/avx512 support")
endif()
if (ENABLE_QPL)
    add_contrib (idxd-config-cmake idxd-config)
    add_contrib (qpl-cmake qpl) # requires: idxd-config
endif ()
[...]
```

With that, we could omit l. 2-11 in contrib/qpl-cmake/CMakeLists.txt and l. 2-13 in contrib/idxd-config/CMakeLists.txt.

File: .gitmodules [hunk 2]
```diff
@@ -338,3 +338,6 @@
 [submodule "contrib/isa-l"]
 	path = contrib/isa-l
 	url = https://github.com/ClickHouse/isa-l.git
+[submodule "contrib/idxd-config"]
```
[the remark this answers is not in the document]
Comment (jinjunzh) [reply .gitmodules]: Done

File: contrib/CMakeLists.txt [hunk 3]
```diff
@@ -178,6 +178,7 @@ add_contrib (sqlite-cmake sqlite-amalgamation)
 add_contrib (s2geometry-cmake s2geometry)
 add_contrib (c-ares-cmake c-ares)
 add_contrib (qpl-cmake qpl)
+add_contrib (idxd-config-cmake idxd-config)
```
[the remark this answers is not in the document]
Comment (jinjunzh) [reply contrib/CMakeLists.txt]: Done

File: contrib/idxd-config-cmake/CMakeLists.txt (hunk 1 above)
Comment (jinjunzh) [reply contrib/idxd-config-cmake/CMakeLists.txt]: Done

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**